packages/generator/src/options/options-parser.ts

```diff
@@ -1,12 +1,62 @@
 import { createLogger } from '@sap-cloud-sdk/util';
-import { Options } from 'yargs';
-import { ParsedOptions } from './generator-options';
+import { InferredOptionType, Options } from 'yargs';
 const logger = createLogger('generator-options');
 
-type Option = Options & {
+/**
+ * @internal
+ * Options for SAP Cloud SDK generators.
+ * Some keys are modified, for more explicit parsing.
+ */
+export type Option = Omit<Options, 'coerce'> & {
+  /**
+   * Name of the option the current option was replaced with.
+   */
   replacedBy?: string;
-  describe: string;
-};
+
+  /**
+   * Only required options should set `demandOption`. Other options are not required by default.
+   */
+  demandOption?: true;
+
+  /**
+   * Coerce function with additional parameter for all options.
+   */
+  coerce?: (arg: any, options: any) => any;
+} & Required<Pick<Options, 'describe' | 'type'>>;
+
+/**
+ * @internal
+ * Helper to represent parsed options based on a public generator options type and a CLI options configuration.
+ * - Makes all properties required.
+ * - Removes deprecated and replaced options.
+ * - Replaces input types with coerced types.
+ * @typeParam GeneratorOptionsOptionsT - Public generator options.
+ * @typeParam CliOptionsT - Configuration of CLI options.
+ */
+export type ParsedOptions<CliOptionsT extends Record<string, Option>> = Omit<
+  { [K in keyof CliOptionsT]: ParsedOptionType<CliOptionsT[K]> },
+  OptionsWith<'deprecated' | 'replacedBy', CliOptionsT>
+>;
+
+type ParsedOptionType<OptionT extends Option> = OptionT extends {
+  coerce: (...arg: any) => infer ReturnT;
```

Comment:
[pp] I think our `coerceRequiredPathArg` and `coercePathArg` have the same signature like (arg,options)=>ReturnT which is always string. Could we type it stronger here?

Reply:
Not sure I understand what you mean.

I could change `(...arg: any) => infer ReturnT` to something like `(arg: any, options: XY) => infer ReturnT;`, but what is the benefit? (and I don't think that is what you meant)

`coerceRequiredPathArg` and `coercePathArg` return `string` and `string | undefined` which is a not always `string`. The coercion functions could however return an arbitrary type (e.g. `string | boolean`), so restricting this to `string` would be wrong in my opinion.

Reply:
discussed in meeting